databricks/koalas/indexing.py

```diff
@@ -492,22 +503,38 @@ def __setitem__(self, key, value):
         rows_sel, cols_sel = key
 
         if (not isinstance(rows_sel, slice)) or (rows_sel != slice(None)):
-            raise SparkPandasNotImplementedError(
-                description="""Can only assign value to the whole dataframe, the row index
-                has to be `slice(None)` or `:`""",
-                pandas_function=".loc[..., ...] = ...",
-                spark_target_function="withColumn, select")
+            if isinstance(rows_sel, list):
+                if isinstance(cols_sel, str):
+                    cols_sel = [cols_sel]
+                kdf = self._kdf
+                sdf = kdf._sdf
+                for col_sel in cols_sel:
+                    sdf = sdf.withColumn(
+                        col_sel,
+                        (F.when(F.col(kdf._internal.index_columns[0]).isin(rows_sel), value)
+                            .otherwise(F.col(col_sel))))
```

Comment:
@itholic, can we use `kdf[...]` to allow operations on different DataFrames? For example,

```python
>>> import pandas as pd
>>> df = pd.DataFrame([[1, 2], [4, 5], [7, 8]],
...                   index=['cobra', 'viper', 'sidewinder'],
...                   columns=['max_speed', 'shield'])
>>> df2 = pd.DataFrame([[1, 2], [4, 5], [7, 8]],
...                   index=['cobra', 'viper', 'sidewinder'],
...                   columns=['max_speed', 'shield'])
>>> df.loc[['viper', 'sidewinder'], ['shield']] = df2.max_speed
>>> df
            max_speed  shield
cobra               1       2
viper               4       4
sidewinder          7       7
```

Reply:
@HyukjinKwon Thanks!! i've learned really a lot. 😍